controllers/imagepolicy_controller.go

```diff
@@ -107,6 +114,21 @@ func (r *ImagePolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
+	// check if we are allowed to use the referenced ImageRepository
+	if _, err := r.hasAccessToRepository(ctx, req, pol.Spec.ImageRepositoryRef, repo.Spec.AccessFrom); err != nil {
+		imagev1.SetImagePolicyReadiness(
+			&pol,
+			metav1.ConditionFalse,
+			"AccessDenied",
```

Comment:
const?

Reply:
I'll leave it as it is, when this code will be factored out into `fluxcd/pkg` the `AccessDenied` reason will be part of the `apis/meta`.

Reply:
OK